orangecontrib/bio/genialis/genapi.py

```diff
@@ -0,0 +1,190 @@
+        """
+        Function downloads all available :obj:`GenData` etc objects from DictyExpress database.
+        Objects are stored in buffer for further use.
+        If buffer is set to None function returns a list of :obj:`GenData` objects.
+        """
+
+        def optcb():
+            if callback:
+                callback()
+
+        cbc = CallBack(1, optcb, callbacks=70)
+
+        store_experiments = []
+
+        list_of_experiments = self._gen.api.data.get(case_ids__contains='5535115cfad58d5e03006217', status='done',
+                                    type__startswith='data:etc:')['objects']
+        cbc.end()
+
+        cbc = CallBack(len(list_of_experiments), optcb, callbacks=30)
+
+        for dict in list_of_experiments:
+            experiment = GenData(dict, self._gen)  # create GenData object
+            store_experiments.append(experiment)
+            cbc()
+
+        cbc.end()
+
+
+        if self.buffer is not None:
+            self.to_buffer(self.email, store_experiments)  # store GenData objects into buffer
+        else:
+            cbc.end()
+            return store_experiments
+
+
+
+    def download_etc_data(self, genData,  callback=None):
+
+        """
+        Function downloads etc data of a chosen experiment from the server. Downloaded data is stored in cache.
+        If data is already in cache than it returns cached data.
+
+        :param genData: :obj:`GenData` object.
+        :return: Returns data in json like format
+        :rtype: :obj:`dict`
+        """
+
+        def optcb():
+            if callback:
+                callback()
+
+        cbc = CallBack(2, optcb, callbacks=50)
+
+        #  If experiment id is in cache, use cached data.
+        if self.buffer is not None:
+            if self.in_buffer(genData.id):
+                cbc.end()
+                return self.from_buffer(genData.id)
+
+        response = genData.download('output.etcfile')
+        cbc()
+        response_gzipped = io.BytesIO(response.content)
+        response_content = io.TextIOWrapper(gzip.GzipFile(fileobj=response_gzipped), encoding="utf-8")
+        cbc()
+        json_object = json.load(response_content)
+
+        self.to_buffer(genData.id, json_object)  # Store data in cache
+        cbc.end()
+
+        return json_object
+
+    def etc_to_table(self, etc_json, callback=None):
+
+        """
+        Converts data from Json to :obj:`Orange.data.table`
+
+
+        :param etc_json: Data in json like format
+        :return: Function returns Orange data table
+        :rtype: :obj:`Orange.data.Table`
+
+        """
+
+        def optcb():
+            if callback:
+                callback()
+
+        cbc = CallBack(2, optcb, callbacks=50)
+
+        json_object = etc_json
+
+        variables = []
+        tp = 1
+        for time in json_object['etc']['timePoints']:
+            var = ContinuousVariable("TP " + str(tp))
+            var.attributes["Time"] = float(time)
```

Comment:
In general you should decide for one type of quotes and always use them (unless you have a very good reason to use the other). Most people use single quotes. Double quotes for user facing string is an exception popular in web development. For instance, this can help you find them later if you decide to support another language.